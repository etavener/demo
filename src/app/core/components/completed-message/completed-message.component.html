<div>
  <div>Your payment has now been authorized.</div>
</div>
