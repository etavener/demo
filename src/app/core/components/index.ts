export * from './card-form/card-form.component';
export * from './completed-message/completed-message.component';
export * from './payment-info/payment-info.component';
export * from './progress-display/progress-display.component';
export * from './input-expiry-date/input-expiry-date.component';
