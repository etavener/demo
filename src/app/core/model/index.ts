export * from './payment';
export * from './worldpayResponse';
