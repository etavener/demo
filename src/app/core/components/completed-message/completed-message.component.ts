import { Component } from '@angular/core';

@Component({
  selector: 'completed-message',
  templateUrl: './completed-message.component.html',
  styleUrls: ['./completed-message.component.scss']
})
export class CompletedMessageComponent {

  constructor() { }

}
