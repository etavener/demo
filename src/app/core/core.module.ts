import { NgModule } from '@angular/core';
import {
  CardFormComponent,
  PaymentInfoComponent,
  CompletedMessageComponent,
  ProgressDisplayComponent,
  InputExpiryDateComponent,
} from './components';

import {
  DateAdapter, MAT_DATE_FORMATS, MAT_DATE_LOCALE,
  MatButtonModule,
  MatDatepickerModule,
  MatDividerModule,
  MatFormFieldModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule,
  MatSelectModule,
  MatStepperModule
} from '@angular/material';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterModule} from '@angular/router';
import { STEPPER_GLOBAL_OPTIONS } from '@angular/cdk/stepper';
import { HttpClientModule } from '@angular/common/http';
import { MomentDateAdapter } from '@angular/material-moment-adapter';
import {WorldpayService} from './services/worldpay.service';
import {BrowserAnimationsModule} from '@angular/platform-browser/animations';

export const MY_FORMATS = {
  parse: {
    dateInput: 'MM/YYYY',
  },
  display: {
    dateInput: 'MM/YYYY',
    monthYearLabel: 'MMM YYYY',
    dateA11yLabel: 'LL',
    monthYearA11yLabel: 'MMMM YYYY',
  },
};

const DECLARATIONS = [
  CardFormComponent,
  PaymentInfoComponent,
  CompletedMessageComponent,
  ProgressDisplayComponent,
  InputExpiryDateComponent
];

@NgModule({
  declarations: [
    ...DECLARATIONS
  ],
  exports: [
    ...DECLARATIONS
  ],
  imports: [
    FormsModule,
    ReactiveFormsModule,
    RouterModule,
    MatStepperModule,
    MatIconModule,
    MatDatepickerModule,
    MatInputModule,
    MatFormFieldModule,
    MatProgressSpinnerModule,
    MatSelectModule,
    MatButtonModule,
    MatDividerModule,
    HttpClientModule,
    BrowserAnimationsModule
  ],
  providers: [
    WorldpayService,
    {
      provide: STEPPER_GLOBAL_OPTIONS,
      useValue: { displayDefaultIndicatorType: false }
    },
    {
      provide: DateAdapter,
      useClass: MomentDateAdapter,
      deps: [MAT_DATE_LOCALE]},
    {
      provide: MAT_DATE_FORMATS,
      useValue: MY_FORMATS
    },
  ]
})
export class CoreModule { }
