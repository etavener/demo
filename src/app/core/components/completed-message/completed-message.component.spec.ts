import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { CompletedMessageComponent } from './completed-message.component';

describe('CompletedMessageComponent', () => {
  let component: CompletedMessageComponent;
  let fixture: ComponentFixture<CompletedMessageComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ CompletedMessageComponent ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CompletedMessageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(fixture).toMatchSnapshot();
  });
});
