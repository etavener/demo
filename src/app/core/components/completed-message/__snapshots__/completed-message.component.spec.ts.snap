// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompletedMessageComponent should create 1`] = `
<completed-message>
  <div>
    <div>
      Your payment has now been authorized.
    </div>
  </div>
</completed-message>
`;
